pkg/controller/controller.go

```diff
@@ -426,7 +430,8 @@ func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 	}
 
 	<-stopCh
-	glog.Info("Shutting down Machine Controller Manager ")
+	glog.V(1).Info("Shutting down Machine Controller Manager ")
+	handlers.UpdateHealth(false)
```

Comment:
Is it possible that just after health status is updated here, liveness probe fails, and the container is restarted before all the `defer` statements execute.

Reply:
The container takes 30s before restarting, and this is a very rare scenario to get to this as, the controller doesn't actually terminate by itself. So i think this should be fine.

Reply:
I am just afraid that this will introduce some indeterministic behavior unnecessarily, however rare it (killing of container when shutting down because of health failure) might be.
Approving the PR.